recipe_scrapers/ottolenghi.py

```diff
@@ -0,0 +1,69 @@
+from ._abstract import AbstractScraper
+from ._grouping_utils import group_ingredients
+
+
+class Ottolenghi(AbstractScraper):
+    @classmethod
+    def host(cls):
+        return "ottolenghi.co.uk"
+
+    def author(self):
+        return self.schema.author()
+
+    def title(self):
+        return self.schema.title()
+
+    def category(self):
+        return self.schema.category()
+
+    def total_time(self):
+        return self.schema.total_time()
+
+    def image(self):
+        return self.soup.find("div", class_="c-recipe-header__gallery").find("img")[
+            "src"
+        ]
+
+    def ingredients(self):
+        return self.schema.ingredients()
+
+    def ingredient_groups(self):
+        return group_ingredients(
+            self.ingredients(),
+            self.soup,
+            ".c-recipe-ingredients__heading",
+            ".c-recipe-ingredients tr:not(:has(.c-recipe-ingredients__heading))",
+        )
+
+    def instructions(self):
+        return self.schema.instructions()
+
+    def ratings(self):
+        return self.schema.ratings()
+
+    def cuisine(self):
+        return self.schema.cuisine()
+
+    def description(self):
+        return self.schema.title()
+
+    def yields(self):
+        return (
+            self.soup.find("div", class_="c-recipe-header__timings")
+            .find("span")
+            .get_text(strip=True)
+        )
+
+    def prep_time(self):
+        return (
+            self.soup.find("div", class_="c-recipe-header__timings")
+            .find_all("span")[1]
+            .get_text(strip=True)
+        )
+
+    def cook_time(self):
+        return (
+            self.soup.find("div", class_="c-recipe-header__timings")
+            .find_all("span")[2]
+            .get_text(strip=True)
+        )
```

Comment:
```suggestion
    def _extract_timing_elements(self, prefix):
        timings = self.soup.find("div", class_="c-recipe-header__timings").find_all("span")
        for timing in timings:
            if prefix.lower() in timing.get_text().lower():
                return timing.get_text()

    def yields(self):
        yield_text = self._extract_timing_elements("serves")
        return get_yields(yield_text)

    def prep_time(self):
        prep_text = self._extract_timing_elements("prep")
        return get_minutes(prep_text)

    def cook_time(self):
        cook_text = self._extract_timing_elements("cook")
        if cook_text and '5o' in cook_text:
            cook_text = cook_text.replace('5o', '50')
        return get_minutes(cook_text)
```

What do you think about refactoring this to use a shared helper and using the starting text as the matching parameter instead of position?

Also I implemented two existing utils `get_minutes` and `get_yields` to normalize the fields outputs. this will require some changes to the test JSONs

Reply:
Also in `cook_time` I added coverage for an error on the recipe page where **50** is displayed as **5o**